---
title: "Old INBRE Homework"
output: html_document
date: "2023-03-31"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

## Installing the rentrez package

The first step is to load the package before you can use the commands. To do this use the following command:

```{r}
install.packages("rentrez, dependencies=TRUE")
```
## Loading rentrez

Use the library(rentrz) command to load the package and to begin using it

```{r}
library(rentrez)
# Warning: package 'rentrez' was built under R version 3.2.5
```
if the only waringin you get is related to the version of R in which the package was built, you should be ready togo

## Databases

You should now be ready to list the NCBI databases

```{r}
entrez_dbs()
##  [1] "pubmed"          "protein"         "nuccore"        
##  [4] "nucleotide"      "nucgss"          "nucest"         
##  [7] "structure"       "sparcle"         "genome"         
## [10] "annotinfo"       "assembly"        "bioproject"     
## [13] "biosample"       "blastdbinfo"     "books"          
## [16] "cdd"             "clinvar"         "clone"          
## [19] "gap"             "gapplus"         "grasp"          
## [22] "dbvar"           "gene"            "gds"            
## [25] "geoprofiles"     "homologene"      "medgen"         
## [28] "mesh"            "ncbisearch"      "nlmcatalog"     
## [31] "omim"            "orgtrack"        "pmc"            
## [34] "popset"          "probe"           "proteinclusters"
## [37] "pcassay"         "biosystems"      "pccompound"     
## [40] "pcsubstance"     "pubmedhealth"    "seqannot"       
## [43] "snp"             "sra"             "taxonomy"       
## [46] "unigene"         "gencoll"         "gtr"
```

## Database Specs

For version information on any database, use the entrez_db_summary() command. First let's ask for help on that command.

```{r}
help("entrez_db_summary")

## starting httpd help server ...

## done
```
This will pull up the manual for that command. Now lets examine the nucleotide database
```{r}
entrez_db_summary("nucleotide")
##  DbName: nuccore
##  MenuName: Nucleotide
##  Description: Core Nucleotide db
##  DbBuild: Build170116-1955m.1
##  Count: 226105014
##  LastUpdate: 2017/01/17 08:16
```

## Database Searchable Fields

Let's say you want to list all fields that can be searched in a specific database. Use the entrez_db_searchable() command. Let's try it for the nucleotide database.

```{r}
entrez_db_searchable("nucleotide")
## Searchable fields for database 'nuccore'
##   ALL     All terms from all searchable fields 
##   UID     Unique number assigned to each sequence 
##   FILT    Limits the records 
##   WORD    Free text associated with record 
##   TITL    Words in definition line 
##   KYWD    Nonstandardized terms provided by submitter 
##   AUTH    Author(s) of publication 
##   JOUR    Journal abbreviation of publication 
##   VOL     Volume number of publication 
##   ISS     Issue number of publication 
##   PAGE    Page number(s) of publication 
##   ORGN    Scientific and common names of organism, and all higher levels of taxonomy 
##   ACCN    Accession number of sequence 
##   PACC    Does not include retired secondary accessions 
##   GENE    Name of gene associated with sequence 
##   PROT    Name of protein associated with sequence 
##   ECNO    EC number for enzyme or CAS registry number 
##   PDAT    Date sequence added to GenBank 
##   MDAT    Date of last update 
##   SUBS    CAS chemical name or MEDLINE Substance Name 
##   PROP    Classification by source qualifiers and molecule type 
##   SQID    String identifier for sequence 
##   GPRJ    BioProject 
##   SLEN    Length of sequence 
##   FKEY    Feature annotated on sequence 
##   PORG    Scientific and common names of primary organism, and all higher levels of taxonomy 
##   COMP    Component accessions for an assembly 
##   ASSM    Assembly 
##   DIV     Division 
##   STRN    Strain 
##   ISOL    Isolate 
##   CULT    Cultivar 
##   BRD     Breed 
##   BIOS    BioSample
```
## Searching a database

To search a database, use the entrez_search() command. Let's examine the manual

```{r}
help("entrez_search")
```
